# Question 1

In [1]:
import mlflow
print(mlflow.__version__)

2.3.2


# Question 2

In [6]:
!python ./homework/preprocess_data.py --raw_data_path ./data --dest_path ./output

In [8]:
!ls -l ./output | grep dv.pkl

-rw-r--r--  1 qavila  staff   153660 May 25 11:48 dv.pkl


In [9]:
import subprocess

command = "ls -l ./output | grep dv.pkl"
output = subprocess.check_output(command, shell=True).decode().strip().split()
file_size = output[4]

print(f"File size: {file_size} bytes")

File size: 153660 bytes


# Question 3

rf = RandomForestRegressor(max_depth=10, random_state=0)

In [10]:
!python ./homework/train.py

2023/05/25 14:08:57 INFO mlflow.tracking.fluent: Experiment with name 'homework' does not exist. Creating a new experiment.


# Question 4

In [13]:
!python ./homework/hpo.py

[I 2023-05-25 14:24:18,545] A new study created in memory with name: no-name-0d058a40-b2e5-4da3-adbe-97651b7eb09d
[I 2023-05-25 14:24:21,069] Trial 0 finished with value: 2.451379690825458 and parameters: {'n_estimators': 25, 'max_depth': 20, 'min_samples_split': 8, 'min_samples_leaf': 3}. Best is trial 0 with value: 2.451379690825458.
[I 2023-05-25 14:24:21,378] Trial 1 finished with value: 2.4667366020368333 and parameters: {'n_estimators': 16, 'max_depth': 4, 'min_samples_split': 2, 'min_samples_leaf': 4}. Best is trial 0 with value: 2.451379690825458.
[I 2023-05-25 14:24:23,680] Trial 2 finished with value: 2.449827329704216 and parameters: {'n_estimators': 34, 'max_depth': 15, 'min_samples_split': 2, 'min_samples_leaf': 4}. Best is trial 2 with value: 2.449827329704216.
[I 2023-05-25 14:24:24,661] Trial 3 finished with value: 2.460983516558473 and parameters: {'n_estimators': 44, 'max_depth': 5, 'min_samples_split': 3, 'min_samples_leaf': 1}. Best is trial 2 with value: 2.44982732

max_depth	15
min_samples_leaf	4
min_samples_split	2
n_estimators	34
n_jobs	-1
random_state	42

RMSE 2.45

# Question 5

In [14]:
!python ./homework/register_model.py

2023/05/25 14:34:24 INFO mlflow.tracking.fluent: Experiment with name 'random-forest-best-models' does not exist. Creating a new experiment.
2023/05/25 14:34:32 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/qavila/.pyenv/versions/3.11.1/envs/practice2/lib/python3.11/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils."
Successfully registered model 'best_model'.
2023/05/25 14:34:58 INFO mlflow.tracking._model_registry.client: Waiting up to 300 seconds for model version to finish creation. Model name: best_model, version 1
Created version '1' of model 'best_model'.


RMSE of best model: 2.45

# Question 6

All of the above